### PR TITLE
[test][sagemaker] Fix coverage issues caused by test deselection

### PR DESCRIPTION
**Pull request by saimidu**
*Issue #, if available:*

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]

## Pytest Marker Checklist
- [x] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [x] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [x] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [x] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [x] I've verified the code change on the config file mentioned above has already been reverted

*Description:*
This PR resolves tests that are skipped due to being accidentally deselected (instead of intentionally deselected)

*Tests run:*

*DLC image/dockerfile:*

*Additional context:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/test_utils/sagemaker.py [hunk 1]
```diff
@@ -139,10 +140,12 @@ def generate_sagemaker_pytest_cmd(image, sagemaker_test_type):
...
-
-    efa_flag = ""
-    efa_dedicated = os.getenv("EFA_DEDICATED", "False").lower() == "true"
-    efa_flag = '--efa' if efa_dedicated else '-m \"not efa\"'
+    if is_pr_context():
+        efa_tests_disabled = os.getenv("DISABLE_EFA_TESTS", "False").lower() == "true"
+        efa_flag = "-m \"not efa\"" if efa_tests_disabled else ""
+    else:
+        efa_dedicated = os.getenv("EFA_DEDICATED", "False").lower() == "true"
+        efa_flag = '--efa' if efa_dedicated else '-m \"not efa\"'
```
Comment (saimidu) [inline test/test_utils/sagemaker.py:148]: This is the actual change in this PR

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_pt_s3_plugin.py [hunk 2]
```diff
@@ -82,4 +82,3 @@ def test_pt_s3_plugin_sm_cpu(sagemaker_session, framework_version, ecr_image):
         )
         job_name = utils.unique_name_from_base('test-pytorch-s3-plugin-cpu')
         pytorch.fit(job_name=job_name)
```
Comment (arjkesh) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_pt_s3_plugin.py:84]: nit: why remove enplane?

File: test/test_utils/sagemaker.py [hunk 3]
```diff
@@ -139,10 +140,12 @@ def generate_sagemaker_pytest_cmd(image, sagemaker_test_type):
     test_report = os.path.join(os.getcwd(), "test", f"{job_type}_{tag}.xml")
     local_test_report = os.path.join(UBUNTU_HOME_DIR, "test", f"{job_type}_{tag}_sm_local.xml")
 
-
-    efa_flag = ""
-    efa_dedicated = os.getenv("EFA_DEDICATED", "False").lower() == "true"
-    efa_flag = '--efa' if efa_dedicated else '-m \"not efa\"'
+    if is_pr_context():
```
Comment (arjkesh) [inline test/test_utils/sagemaker.py:143]: could you provide comments on why there is a diff experience in PR vs otherwise?

**Comment by arjkesh** [comment]
LGTM minus the request for comments

File: test/test_utils/sagemaker.py (hunk 3 above)
Comment (saimidu) [reply test/test_utils/sagemaker.py:143]: Sure. Adding comments now.

**Comment by arjkesh** [comment]
LGTM after comment addressed

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_pt_s3_plugin.py (hunk 2 above)
Comment (saimidu) [reply test/sagemaker_tests/pytorch/training/integration/sagemaker/test_pt_s3_plugin.py:84]: It removed an extra blank line at the end

**Review by arjkesh — APPROVED** [review]

**Pull request merged**